### PR TITLE
Hang - HW7

### PR DESCRIPTION
**Pull request by hanghu**
The crime.py creates a module to return numbers of crime reports around a specific house gps corrdinates within the cutoff distance. Besides, the crime information are also grouped and plotted.

### CODE REVIEW

File: HW7/crime_info.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            if  distance <= cutoff:
+                count += 1
+                count_type.append(crime_type[i])
+            else:
+                pass
+
+    #groupby the different types of crime
+    counted_type = Counter(count_type)
+    plt.style.use('ggplot')
+    fig, ax = plt.subplots(figsize=(12,6))
```
Comment (YangJiang89) [inline HW7/crime_info.py]: white space after comma

File: HW7/crime_info.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import numpy as np
```
Comment (YangJiang89) [inline HW7/crime_info.py:7]: Missing doc string

File: HW7/crime_info.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    plt.style.use('ggplot')
+    fig, ax = plt.subplots(figsize=(12,6))
+    ax.bar(range(len(counted_type)), counted_type.values(), align='center')
+    plt.xticks(range(len(counted_type)), counted_type.keys())
+    title = 'Crime distribution around '+house_name+' within the range of '+str(cutoff)+' miles'
+    plt.suptitle(title)
+    plt.savefig('crime distribution')
+    return count
+
+# showing an example of this function
```
Comment (YangJiang89) [inline HW7/crime_info.py]: Can add
if __name__ == "__main__":

File: HW7/crime_info.py (hunk 1 above)
Comment (YangJiang89) [inline HW7/crime_info.py]: unused variable fig

File: HW7/crime_info.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+house_data = pd.read_csv('../data/kc_house_data.csv')
+crime_data = pd.read_csv('../data/King_County_Sheriff_s_Office.csv')
+house_gps = house_gps = (crime_data.latitude[1561], crime_data.longitude[1561])
+crime_lats = list(crime_data.latitude)
+crime_lons = list(crime_data.longitude)
+crime_gps = list(zip(crime_lats,crime_lons))
```
Comment (YangJiang89) [inline HW7/crime_info.py]: white space after comma

File: HW7/crime_info.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from collections import Counter
```
Comment (YangJiang89) [inline HW7/crime_info.py]: import "from collections import Counter" comes before "import numpy as np"

File: HW7/crime_info.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+from geopy.distance import vincenty
+
+def crime_count(house_gps, crime_gps, crime_type,
+                    cutoff=2.0, house_name='the house'):
```
Comment (MengyuanZoe) [inline HW7/crime_info.py]: wrong continued indentation, remove 4 spaces,

File: HW7/crime_info.py (hunk 1 above)
Comment (MengyuanZoe) [inline HW7/crime_info.py]: add one space after comma

File: HW7/crime_info.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    fig, ax = plt.subplots(figsize=(12,6))
+    ax.bar(range(len(counted_type)), counted_type.values(), align='center')
+    plt.xticks(range(len(counted_type)), counted_type.keys())
+    title = 'Crime distribution around '+house_name+' within the range of '+str(cutoff)+' miles'
+    plt.suptitle(title)
+    plt.savefig('crime distribution')
+    return count
+
+# showing an example of this function
+house_data = pd.read_csv('../data/kc_house_data.csv')
```
Comment (MengyuanZoe) [inline HW7/crime_info.py]: Invalid constant name. Rename.

File: HW7/crime_info.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+crime_data = pd.read_csv('../data/King_County_Sheriff_s_Office.csv')
+house_gps = house_gps = (crime_data.latitude[1561], crime_data.longitude[1561])
+crime_lats = list(crime_data.latitude)
+crime_lons = list(crime_data.longitude)
+crime_gps = list(zip(crime_lats,crime_lons))
+crime_type = list(crime_data['incident_type_primary'])
+house_name = 'House NO. 1561'
+count_house1561 = crime_count(house_gps, crime_gps, crime_type, house_name=house_name)
```
Comment (MengyuanZoe) [inline HW7/crime_info.py]: Invalid constant name. Rename.

File: HW7/crime_info.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from collections import Counter
+from geopy.distance import vincenty
+
+def crime_count(house_gps, crime_gps, crime_type,
+                    cutoff=2.0, house_name='the house'):
+    """
+    Count return numbers of crime reports around a specifc house gps corrdinates
```
Comment (MengyuanZoe) [inline HW7/crime_info.py]: Explain the return value clearer.

File: HW7/crime_info.py (hunk 6 above)
Comment (hanghu) [reply HW7/crime_info.py]: removed spaces

File: HW7/crime_info.py (hunk 1 above)
Comment (hanghu) [reply HW7/crime_info.py]: added comma

File: HW7/crime_info.py (hunk 9 above)
Comment (hanghu) [reply HW7/crime_info.py]: added types of return value as int

**Comment by hanghu** [comment]
improved code with constant renamed, modules docstring added and other style error fixed.
also added "if __name__ = __main__ for the small test part

**Pull request merged**